src/modules/UI/scenes/Plugins/EdgeProvider.tsx

```diff
@@ -398,19 +403,78 @@ export class EdgeProvider extends Bridgeable {
       return
     }
 
-    // Prepare the internal spend request:
-    const info: GuiMakeSpendInfo = {
-      currencyCode,
-      nativeAmount: result.nativeAmount,
-      publicAddress: result.legacyAddress || result.publicAddress,
+    if (currencyCode !== this.selectedCurrencyCode) {
+      throw new Error('URI currency code mismatch from chooseCurrencyWallet selected code')
+    }
+
+    // PUBLIC ADDRESS URI
+    let tokenId: string | undefined
+    if (this.selectedChainCode !== this.selectedCurrencyCode) {
+      tokenId = getTokenId(account, this.selectedWallet.currencyInfo.pluginId, currencyCode)
+    }
+    const spendInfo: EdgeSpendInfo = {
       customNetworkFee,
       metadata,
-      lockInputs,
-      uniqueIdentifier
+      spendTargets: [
+        {
+          // Prioritize legacyAddress first since the existence of a legacy address means that a legacy address
+          // was scanned. Plugins may translate a legacy address into a publicAddress and provide that as well
+          publicAddress: legacyAddress ?? publicAddress,
+          memo: uniqueIdentifier,
+          nativeAmount
+        }
+      ],
+      tokenId
     }
+    return this._requestSpendCommon({ lockInputs, orderId, spendInfo })
+  }
 
-    // Launch:
-    return this._makeSpendRequest(info, this.selectedWallet, orderId, this.selectedCurrencyCode)
+  /**
+   * Internal helper to launch the send confirmation scene.
+   */
+  async _requestSpendCommon({
+    lockInputs,
+    orderId,
+    spendInfo
+  }: {
+    lockInputs: boolean
+    orderId?: string
+    spendInfo: EdgeSpendInfo
+  }): Promise<EdgeTransaction | undefined> {
+    return new Promise((resolve, reject) => {
+      const lockTilesMap = lockInputs ? { address: true, amount: true, wallet: true } : undefined
+
+      this._navigation.push('send2', {
+        walletId: this.selectedWallet.id,
+        spendInfo,
+        lockTilesMap,
+        onBack: () => resolve(undefined),
+        onDone: (error, transaction) => {
+          if (error != null) {
+            reject(error)
+            return
+          }
+          if (transaction == null) {
+            reject(new Error('Missing transaction'))
+            return
+          }
+          this._navigation.pop()
+
+          this.selectedWallet.nativeToDenomination(transaction.nativeAmount, transaction.currencyCode).then(exchangeAmount => {
```

Comment:
This might cause an unhandled promise rejection. Can we put a `.catch` on the end in case something goes wrong with the conversion tracking? It could be a no-op if we really don't care, like `catch(()=>{})`, but we should at least acknowledge it.

Reply:
👍